File: frontend/src/types/cv.ts
/** TypeScript types for CV data structure */

export interface PersonalInfo {
  name: string;
  email?: string;
  phone?: string;
  address?: string;
  linkedin?: string;
  github?: string;
  website?: string;
  summary?: string;
}

export interface Experience {
  title: string;
  company: string;
  start_date: string;
  end_date?: string;
  description?: string;
  location?: string;
}

export interface Education {
  degree: string;
  institution: string;
  year?: string;
  field?: string;
  gpa?: string;
}

export interface Skill {
  name: string;
  category?: string;
  level?: string;
}

export interface CVData {
  personal_info: PersonalInfo;
  experience: Experience[];
  education: Education[];
  skills: Skill[];
}

export interface CVResponse {
  cv_id: string;
  filename?: string;
  status: string;
}

export interface CVListItem {
  cv_id: string;
  created_at: string;
  updated_at: string;
  person_name?: string;
}

export interface CVListResponse {
  cvs: CVListItem[];
  total: number;
}

File: backend/models/cv.py
"""CV data and response models."""
from typing import Optional, List
from pydantic import BaseModel, Field
from backend.models.personal import PersonalInfo
from backend.models.experience import Experience
from backend.models.education import Education, Skill


class CVData(BaseModel):
    """Complete CV data model."""

    personal_info: PersonalInfo
    experience: List[Experience] = []
    education: List[Education] = []
    skills: List[Skill] = []
    theme: Optional[str] = Field(
        default="classic",
        description="CV theme: accented, classic, colorful, creative, elegant, executive, minimal, modern, professional, or tech",
    )


class CVResponse(BaseModel):
    """CV creation response."""

    cv_id: str
    filename: Optional[str] = None
    status: str = "success"


class CVListItem(BaseModel):
    """CV list item."""

    cv_id: str
    created_at: str
    updated_at: str
    person_name: Optional[str] = None
    filename: Optional[str] = None


class CVListResponse(BaseModel):
    """CV list response."""

    cvs: List[CVListItem]
    total: int
